scripts/prism_downloader.py

```diff
@@ -71,63 +121,96 @@ def parseParameters(paramString: str) -> list[str]:
     param_list = paramString.split(',')
     if not param_list[0]:
         return DEFAULT_PARAMS
+    
+    for param in param_list:
+        if param not in DEFAULT_PARAMS:
+            raise ValueError(f"Invalid parameter: {param}. Must be one of {DEFAULT_PARAMS}")
+    
     return param_list
 
-def clean_up_files(files:list) -> None:
+def parseDateRange(startDateString: str, endDateString: str,  frequency: str) -> pd.DatetimeIndex:
+    if frequency == 'daily':
+        start_date = dt.strptime(startDateString, "%Y-%m-%d")
+        end_date = dt.strptime(endDateString, "%Y-%m-%d")
+        freq = '1D'
+        date_range = pd.date_range(start_date, end_date, freq=freq, normalize=True).strftime('%Y%m%d')
+    elif frequency == 'monthly':
+        start_date = dt.strptime(startDateString, "%Y-%m")
+        end_date = dt.strptime(endDateString, "%Y-%m")
+        freq = '1ME'
+        date_range = pd.date_range(start_date, end_date, freq=freq, normalize=True).strftime('%Y%m')
+    elif frequency == 'annual':
+        start_date = dt.strptime(startDateString, "%Y")
+        end_date = dt.strptime(endDateString, "%Y")
+        freq = '1Y'
+        date_range = pd.date_range(start_date, end_date, freq=freq ,normalize=True).strftime('%Y')
+    else:
+        raise ValueError(f"Invalid frequency: {frequency}. Must be one of {FREQUENCY_OPTIONS}")
```

Comment:
See above comment (I think given your argparse choices setup you don't need this) https://docs.python.org/3/library/argparse.html#choices ... although I suppose it doesn't hurt if you want to eventually use the function elsewhere